### PR TITLE
Add support for multiple profiles

### PR DESCRIPTION
**Pull request by stevetemple**
## Describe your changes

Added support for multiple profiles allowing more than one app registration to be used

## Link to issue

https://github.com/Gibe/Umbraco.Community.AzureSSO/issues/21

## Checklist before requesting a review
- [x] I have performed a self-review of my code

### CODE REVIEW

File: src/Umbraco.Community.AzureSSO/MicrosoftAccountBackOfficeExternalLoginProviderOptions.cs [hunk 1]
```diff
@@ -126,5 +128,10 @@ private string DisplayName(ClaimsPrincipal claimsPrincipal, string defaultValue)
 
 			return !string.IsNullOrWhiteSpace(displayName) ? displayName : defaultValue;
 		}
+
+		public void Configure(BackOfficeExternalLoginProviderOptions options)
+		{
+			// TODO : What do we do here
```
Comment (Reaction77) [inline src/Umbraco.Community.AzureSSO/MicrosoftAccountBackOfficeExternalLoginProviderOptions.cs:134]: I think we should be maintaining the previous behaviour here, or if that isn't possible then let's get rid of this redundant method.

File: src/Umbraco.Community.AzureSSO/MicrosoftAccountAuthenticationExtensions.cs [hunk 2]
```diff
@@ -16,31 +18,61 @@ internal static IUmbracoBuilder AddMicrosoftAccountAuthentication(this IUmbracoB
...
+							backOfficeAuthenticationBuilder =>
+							{
+								backOfficeAuthenticationBuilder.AddMicrosoftIdentityWebApp(options =>
+										{
+											CopyCredentials(options, profile.Credentials);
+											options.SignInScheme = backOfficeAuthenticationBuilder.SchemeForBackOffice(profile.Name);
+											options.Events = new OpenIdConnectEvents();
+											
+										},
+										displayName: profile.DisplayName ?? "Azure Active Directory",
```
Comment (Reaction77) [inline src/Umbraco.Community.AzureSSO/MicrosoftAccountAuthenticationExtensions.cs:40]: Should we keep up with Microsoft's (silly) renaming of AAD to "Microsoft Entra ID"?

**Review by Reaction77 — CHANGES_REQUESTED** [review]
Just concerned about version compatibility/documentation. Do we want to force all usages to use "Profiles" or do we want to still allow the previous configuration to work also? From what I can tell the Construction of AzureSsoSettings is forcing the configuration into a profile.

File: src/Umbraco.Community.AzureSSO/MicrosoftAccountBackOfficeExternalLoginProviderOptions.cs (hunk 1 above)
Comment (stevetemple) [reply src/Umbraco.Community.AzureSSO/MicrosoftAccountBackOfficeExternalLoginProviderOptions.cs:134]: Was just thinking on this. It's enforced by the interface, but doesn't appear to be used in this config

**Comment by stevetemple** [comment]
> Just concerned about version compatibility/documentation. Do we want to force all usages to use "Profiles" or do we want to still allow the previous configuration to work also? From what I can tell the Construction of AzureSsoSettings is forcing the configuration into a profile.

I've made it support old config for backwards compatibility. It just makes the exsting config a single profile

**Review by Reaction77 — APPROVED** [review]
A thought about letting consumers know that the Configure method they may have previously used no longer does anything.

File: src/Umbraco.Community.AzureSSO/MicrosoftAccountBackOfficeExternalLoginProviderOptions.cs [hunk 3]
```diff
@@ -126,5 +128,10 @@ private string DisplayName(ClaimsPrincipal claimsPrincipal, string defaultValue)
 
 			return !string.IsNullOrWhiteSpace(displayName) ? displayName : defaultValue;
 		}
+
+		public void Configure(BackOfficeExternalLoginProviderOptions options)
+		{
+			// Unused
```
[the remark this answers is not in the document]
Comment (jdurcan) [reply src/Umbraco.Community.AzureSSO/MicrosoftAccountBackOfficeExternalLoginProviderOptions.cs:134]: Yeah, agreed

[the remark this answers is not in the document]
Comment (stevetemple) [reply src/Umbraco.Community.AzureSSO/MicrosoftAccountBackOfficeExternalLoginProviderOptions.cs:134]: I can do, it won't hurt. However, I did not do it it's not something someone consuming the module would be using, it's setup in our initialisation and not really able to be used by anything else

**Pull request merged**